gitoxide-core/src/hours/util.rs

```diff
@@ -53,9 +52,23 @@ impl WorkByPerson {
     ) -> std::io::Result<()> {
         writeln!(
             out,
-            "{} <{}>",
-            self.name.iter().join(", "),
-            self.email.iter().join(", ")
+            "{names} <{mails}>",
+            names = self
+                .name
+                .iter()
+                // BStr does not impl slice::Join
+                .map(|s| s.to_string())
+                .collect::<Vec<_>>()
```

Comment:
I think you could write an iterator for `BStr` to avoid this allocation (and the `.to_string()`)?

Given that gix uses `BStr` a lot, I think it might worth the trouble.

Reply:
I was not yet sure what the precise missing bound was as to why BStr does not impl Join. Given that the results are probably short and are directly written to the user it seemed fine to convert them to Strings but that might be wrong (?).

Reply:
> I was not yet sure what the precise missing bound was as to why BStr does not impl Join.

`std::slice::join` only accept types that `impl Borrow<str>`, `BStr` doesn't implement it.

> Given that the results are probably short and are directly written to the user it seemed fine to convert them to Strings but that might be wrong (?).

True, though I still think achieving zero-allocation is better.

It probably won't take that much effect given that [`BStr` implements `std::fmt::Display`](https://docs.rs/bstr/latest/bstr/struct.BStr.html#impl-Display-for-BStr), so basically you just need to re-implement join from itertool.

Reply:
Yeah, I inlined the join impl which was fairly compact.